flang/lib/Lower/ConvertExprToHLFIR.cpp

```diff
@@ -1433,6 +1433,19 @@ class HlfirBuilder {
 private:
   hlfir::EntityWithAttributes
   gen(const Fortran::evaluate::BOZLiteralConstant &expr) {
+    mlir::Location loc = getLoc();
+    using Type =
+        Fortran::evaluate::Type<Fortran::evaluate::TypeCategory::Integer, 16>;
+    auto constant = Fortran::evaluate::Constant<Type>{
+        Fortran::evaluate::Scalar<Type>::ConvertUnsigned(std::move(expr))
+            .value};
```

Comment:
You should just forward to `gen(const Fortran::evaluate::Constant<Type>&)` after that rather than dealing with the packaging again.

Reply:
I thought it was a bit clunky. ;)